flink-ml-lib/src/test/java/org/apache/flink/ml/evaluation/BinaryClassificationEvaluatorTest.java

```diff
@@ -0,0 +1,245 @@
+
+    private static final List<Row> INPUT_DATA_DOUBLE_RAW =
+            Arrays.asList(
+                    Row.of(1, 0.9),
+                    Row.of(1, 0.8),
+                    Row.of(1, 0.7),
+                    Row.of(0, 0.75),
+                    Row.of(0, 0.6),
+                    Row.of(1, 0.65),
+                    Row.of(1, 0.55),
+                    Row.of(0, 0.4),
+                    Row.of(0, 0.3),
+                    Row.of(1, 0.35),
+                    Row.of(0, 0.2),
+                    Row.of(1, 0.1));
+
+    private static final List<Row> INPUT_DATA_WITH_MULTI_SCORE =
+            Arrays.asList(
+                    Row.of(1.0, Vectors.dense(0.1, 0.9)),
+                    Row.of(1.0, Vectors.dense(0.1, 0.9)),
+                    Row.of(1.0, Vectors.dense(0.1, 0.9)),
+                    Row.of(0.0, Vectors.dense(0.25, 0.75)),
+                    Row.of(0.0, Vectors.dense(0.4, 0.6)),
+                    Row.of(1.0, Vectors.dense(0.1, 0.9)),
+                    Row.of(1.0, Vectors.dense(0.1, 0.9)),
+                    Row.of(0.0, Vectors.dense(0.6, 0.4)),
+                    Row.of(0.0, Vectors.dense(0.7, 0.3)),
+                    Row.of(1.0, Vectors.dense(0.1, 0.9)),
+                    Row.of(0.0, Vectors.dense(0.8, 0.2)),
+                    Row.of(1.0, Vectors.dense(0.9, 0.1)));
+
+    private static final List<Row> INPUT_DATA_WITH_WEIGHT =
+            Arrays.asList(
+                    Row.of(1.0, Vectors.dense(0.1, 0.9), 0.8),
+                    Row.of(1.0, Vectors.dense(0.1, 0.9), 0.7),
+                    Row.of(1.0, Vectors.dense(0.1, 0.9), 0.5),
+                    Row.of(0.0, Vectors.dense(0.25, 0.75), 1.2),
+                    Row.of(0.0, Vectors.dense(0.4, 0.6), 1.3),
+                    Row.of(1.0, Vectors.dense(0.1, 0.9), 1.5),
+                    Row.of(1.0, Vectors.dense(0.1, 0.9), 1.4),
+                    Row.of(0.0, Vectors.dense(0.6, 0.4), 0.3),
+                    Row.of(0.0, Vectors.dense(0.7, 0.3), 0.5),
+                    Row.of(1.0, Vectors.dense(0.1, 0.9), 1.9),
+                    Row.of(0.0, Vectors.dense(0.8, 0.2), 1.2),
+                    Row.of(1.0, Vectors.dense(0.9, 0.1), 1.0));
+
+    private static final double[] EXPECTED_DATA =
+            new double[] {0.7691481137909708, 0.3714285714285714, 0.6571428571428571};
+    private static final double[] EXPECTED_DATA_M =
+            new double[] {
+                0.8571428571428571, 0.9377705627705628, 0.8571428571428571, 0.6488095238095237
+            };
+    private static final double EXPECTED_DATA_W = 0.8911680911680911;
+    private static final double EPS = 1.0e-5;
+
+    @Before
+    public void before() {
+        Configuration config = new Configuration();
+        config.set(ExecutionCheckpointingOptions.ENABLE_CHECKPOINTS_AFTER_TASKS_FINISH, true);
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(3);
```

Comment:
Thanks for fixing the problem here. Could we also add a test to verify that the algorithm can still work even if parallelism is larger than number of data? `LogisticRegressionTest.testMoreSubtaskThanData` is a good example. Or reviewers would have to manually change this to `env.setParallelism(20)` and re run the tests.